### ISSUE TITLE
Getting error when trying to delete: self.db.deleteByQuery is not a function

### ISSUE THREAD

**Issue by juanpujol** [strongloop-community/loopback-connector-elastic-search#45]
I'm using ElasticSarch 2.1.1 and loopback-connector-elastic-search 1.0.5 and every time I try to delete a record by id I get this error:

```
{  "error": {
    "name": "TypeError",
    "status": 500,
    "message": "self.db.deleteByQuery is not a function",
    "stack": "TypeError: self.db.deleteByQuery is not a function\n    at destroyAll (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-connector-es/lib/esConnector.js:796:13)\n    at doDelete (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/dao.js:1936:19)\n    at /Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/dao.js:1910:9\n    at doNotify (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:98:49)\n    at doNotify (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:98:49)\n    at doNotify (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:98:49)\n    at doNotify (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:98:49)\n    at Function.ObserverMixin._notifyBaseObservers (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:121:5)\n    at Function.ObserverMixin.notifyObserversOf (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:96:8)\n    at Function.ObserverMixin._notifyBaseObservers (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:119:15)\n    at Function.ObserverMixin.notifyObserversOf (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:96:8)\n    at Function.ObserverMixin._notifyBaseObservers (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:119:15)\n    at Function.ObserverMixin.notifyObserversOf (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:96:8)\n    at Function.ObserverMixin._notifyBaseObservers (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:119:15)\n    at Function.ObserverMixin.notifyObserversOf (/Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/observer.js:96:8)\n    at /Users/juan/Documents/Projects/bkn-buildings-api/node_modules/loopback-datasource-juggler/lib/dao.js:1908:13"
  }
}
```

An important issue since I'm not able to remove anything.

Thank you.

**Comment by pulkitsinghal** [comment]
Ah, dang! Its deprecated in 1.5.3

<img width="299" alt="screen shot 2014-12-28 at 11 43 18 am" src="https://cloud.githubusercontent.com/assets/821963/17454982/9e8d231c-5b6f-11e6-9327-6fca51aab963.png">

**Comment by pulkitsinghal** [comment]
Latest stable version of ES is `2.3.4` as of now and that may be made to work according to [this comment] (https://github.com/elastic/elasticsearch/issues/7052#issuecomment-124045013):
> delete-by-query has been implemented using bulk, and moved into a plugin: https://github.com/elastic/elasticsearch/tree/master/plugins/delete-by-query

But I think the `master` version of ES tells a different story for the future completely:
> The Delete-By-Query plugin has been removed in favor of a new Delete By Query API implementation in core. It now supports throttling, retries and cancellation but no longer supports timeouts. Instead use the cancel API to cancel deletes that run too long.

**Comment by juanpujol** [comment]
I added a conditional to the `destroyAll` method so it will work with the native `delete` using just id from ElasticSearch v2.x.

https://gist.github.com/juanpujol/3a3ad4aa8a60b649f632127bd5814d2f#file-esconnector_destroyall-js-L29

It should work as long the `apiVersion` setting is set to `2.x`.

This works for me but I don't know if should go on a PR.

**Comment by pulkitsinghal** [comment]
Btw, it only seems appropriate that I thank both:

1. @arielfr for https://github.com/arielfr/elastic-deletebyquery, 
2. and @dermidgen https://github.com/dermidgen/elasticsearch-deletebyquery

for their separate and novel approach ... without them my research into a possible resolution would not get as far.

**Comment by pulkitsinghal** [comment]
@juanpujol - I like what you have and I do think it would have made a valid PR, even if there may have been a little bit of back & forth.

Right now I'm trying to use https://github.com/arielfr/elastic-deletebyquery to make it work, a bit stuck but let's hope I figure it out :)

**Comment by pulkitsinghal** [comment]
@juanpujol - it is done and released in 1.0.7 of this connector, please take it for a spin and reopen this issue if it is not fixed.

Out of the many approaches, I chose to use https://github.com/arielfr/elastic-deletebyquery to make it work which means that scam/scrol is used in tandem with bulk queries to ES v2.x

**Comment by juanpujol** [comment]
@pulkitsinghal sorry it took me so long. I tested and it works, thank you!
